server/app/assets/javascripts/main.ts

```diff
@@ -303,6 +304,34 @@ export function init() {
     })
   }
 
+  // Note that this formatting logic mimics QuestionDefinition.getQuestionNameKey()
+  const formatQuestionName = (unformatted: string) => {
+    const formatted = unformatted
+      .toLowerCase()
+      .replace(/[^a-zA-Z ]/g, '')
+      .replace(/\s/g, '_')
+    return formatted
+  }
+
+  // Give a live preview of how the question name will be formatted in exports
+  const questionNameInput = document.getElementById('question-name-input')
```

Comment:
```suggestion
  const questionNameInput: HTMLInputElement | null = document.getElementById('question-name-input')
```

Are you able to add the type annotation here like you have for `formattedOutput`? That should negate the need for casting `questionNameInput as HTMLInputElement` further down.

Reply:
Good catch.  I'll update that.

Reply:
Tried setting the type on `questionNameInput` when it's declared as you suggested, but I got this error:  
![image](https://github.com/civiform/civiform/assets/134556723/63e8b7e9-cdb3-465d-a7c2-952ad7303819)
So, I think casting is my only option.  Unless you can think of another way around this?

Reply:
Oh bummer. No worries. It's fine as is then.